### PR TITLE
Add Android build target

### PR DESCRIPTION
**Pull request by jasuarez**
This PR enables to have an "android" build target that allows to compile the DRI drivers for Android.

It's based on Rob Herring work for Linaro (https://ci.linaro.org/job/robher-aosp).

This fixes issue #11.

### CODE REVIEW

File: Rockerfile.mesa [hunk 1]
```diff
@@ -29,31 +31,73 @@ FROM {{ $image }}:llvm-{{ .LLVM }}
...
 
-RUN apt-get update                                   \
-  && apt-get -y --no-install-recommends install git  \
-     {{ if eq .BUILD "windows" }}mingw-w64{{ end }}  \
+RUN apt-get update                                      \
+  && apt-get -y --no-install-recommends install git     \
+     {{ if eq .BUILD "windows" }}mingw-w64{{ end }}     \
+     {{ if eq .BUILD "android" }}                       \
+       wget openjdk-8-jdk python sudo make gcc gettext  \
+       python-mako bison libc6-dev xz-utils             \
```
Comment (tanty) [inline Rockerfile.mesa:39]: I suppose it doesn't matter much but it feels like wget, python, sudo, make gcc, gettext, python-mako, bison, libc6-dev and xz-utils should be already installed, aren't they?

File: Rockerfile.mesa [hunk 2]
```diff
@@ -29,31 +31,73 @@ FROM {{ $image }}:llvm-{{ .LLVM }}
...
+MOUNT {{ $ccachedir }}:/home/local/.ccache:Z
+
+{{ if eq .BUILD "android" }}
+RUN wget https://storage.googleapis.com/git-repo-downloads/repo \
+  && chmod u+x repo                                             \
+  && mkdir -p /home/local/.bin                                  \
+  && mv repo /home/local/.bin
+
+ENV PATH=/usr/lib/ccache:/home/local/.bin/:$PATH
+ENV USE_CCACHE=1
```
Comment (tanty) [inline Rockerfile.mesa:64]: I think we should move to the same point where the MOUNT for ccache is done:

ENV PATH=/usr/lib/ccache:$PATH
ENV USE_CCACHE=1

and leave only the /home/local/.bin and the /usr/lib/llvm-{{ $llvm_version }}/bin to the different branches of the "if"

Also, notice that the USE_CCACHE=1 is only exported in the Android branch (AFAIS)

**Review by tanty — APPROVED** [review]
Other than those 2 nitpicks, this is :+1: :smiley:

File: Rockerfile.mesa (hunk 1 above)
Comment (jasuarez) [reply Rockerfile.mesa:39]: Not really. For Android we are using as base system Ubuntu Xenial, not our base. Reason is that base contains a lot of dependencies that are not useful at lot for Android, and thus we can reduce the final size.

File: Rockerfile.mesa (hunk 2 above)
Comment (jasuarez) [reply Rockerfile.mesa:64]: Sure. I'll move it.

OTOH, yeah, USE_CCACHE is an envvar that Android uses to enable/disable ccache. By default, in Android ccache is disabled, thus we need to explicitly enable here.

Comment (jasuarez) [reply Rockerfile.mesa:64]: Thinking more about it, I think I'll leave as it is. The thing is that I want to ensure that ccache is always the first in the PATH. If I set it at the beginning and later set local/.bin or llvm-bin (depending if it is Android or no) I don't know how to put it after the ccache path.

I'll push as it is, and if later we find a way to fix it, we will do.

**Comment by jasuarez** [comment]
This fixes issue #23

**Pull request merged**